torchvision/csrc/io/decoder/sync_decoder.cpp

```diff
@@ -19,17 +19,17 @@ void SyncDecoder::AVByteStorage::ensure(size_t n) {
 }
 
 uint8_t* SyncDecoder::AVByteStorage::writableTail() {
-  CHECK_LE(offset_ + length_, capacity_);
+  TORCH_CHECK_LE(offset_ + length_, capacity_);
   return buffer_ + offset_ + length_;
 }
 
 void SyncDecoder::AVByteStorage::append(size_t n) {
-  CHECK_LE(n, tail());
+  TORCH_CHECK_LE(n, tail());
```

Comment:
This is not just a check. This is a function call that must be implemented and then checked for its output. Unfortunately the new `TORCH_CHECK_LE` mechanism can skip completely the method call if not in debug mode. This means that the function call will never be done causing issues.